test/pthread/test_pthread_kill.c

```diff
@@ -13,52 +13,58 @@
 #include <signal.h>
 
 _Atomic int sharedVar = 0;
+_Atomic int gotTermSignal = 0;
 
-static void *thread_start(void *arg) {
-  // As long as this thread is running, keep the shared variable latched to nonzero value.
-  for (;;) {
-    ++sharedVar;
+pthread_t thr;
+
+void signal_handler(int sig, siginfo_t * info, void * arg) {
+  printf("signal: %d onthread=%d\n", sig, pthread_self() == thr);
+  if (sig == SIGTERM) {
+    gotTermSignal = 1;
   }
+}
 
-  pthread_exit(0);
+void setup_handler() {
+  struct sigaction act;
+  sigemptyset(&act.sa_mask);
+  act.sa_flags = SA_SIGINFO;
+  act.sa_sigaction = signal_handler;
+  sigaction(SIGTERM, &act, NULL);
 }
 
-void BusySleep(long msecs) {
+
+void sleepms(long msecs) {
   usleep(msecs * 1000);
 }
 
+void *thread_start(void *arg) {
+  // As long as this thread is running, keep the shared variable latched to nonzero value.
+  while (!gotTermSignal) {
+    sleepms(1);
```

Comment:
Why was this sleep added to the test?

Reply:
In order to cancel a running thread the threads needs to reach some cancellation point, in this case the cancellation point is the mailbox check where this thread runs the signal handler code.

Reply:
How did it work before? If it failed before, does that mean users might notice this change?

Reply:
Ah yes, it is possible that this change is observable.  Previously we used `worker.terminate()` to cancel the running thread, but now we require the thread to cooperate somewhat.  It would take a very uncoperative thread indeed to observe this though.  Basically a busy loop like this.

If you prefer I can seperate this change in to just the `pthread_kill` part, and keep the `pthread_cancel` behaviour the same (i.e. keep using `worker.terminate()`).

Reply:
Having said that this test is simply wrong since it was using `pthread_kill` to try to kill a thread.. since is not what that API does.   The public `pthread_kill` API will always being down the entire process.  So this test simply has the wrong expectation of how the API works.

Reply:
Actually I'm wrong.. pthread_kill was implement via `worker.terminate()` but this was always wrong.  `pthread_cancel` has always requires cooperation from the target there.

Thus the only folks who could observe this change are ones (like this test) that were wrongly assuming that pthread_kill could be used to terminate single thread.

Thus, I think this change is good as is.

Reply:
Change sounds good, then. But maybe add a changelog mention?